File: product.py
from mongoengine import Document, StringField, FloatField, ListField, BooleanField
from app.models.schemas import ProductSchema

class Product(Document):
    product_name = StringField(required=True)
    price = FloatField(required=True)
    special_price = FloatField(default=0.0)
    product_description = StringField()
    category = StringField(required=True)
    is_available = BooleanField(default=True)
    product_urls = ListField(StringField())

    def serialize(self):
        product_schema = ProductSchema()
        return product_schema.dump(self)

File: types/product.ts
export interface Product {
  id: string;
  product_name: string;
  product_description: string;
  category: string;
  price: number;
  special_price: number;
  is_available: boolean;
  product_urls: string[];
}

export interface ProductResponse {
  data: Product;
  message: string | null;
  status: string;
}

export interface ProductsListResponse {
  data: {
    products: Product[];
    total_pages?: number;
    current_page?: number;
  };
  message: string | null;
  status: string;
}

export interface CartResponse {
  data: {
    cart: Product[];
  };
  message: string | null;
  status: string;
}